src/lib.rs

```diff
@@ -706,6 +726,94 @@ where
         }
     }
 
+    /// Removes multiple of a value from the bag. If `quantity` is greater than the number of
+    /// occurences, zero occurances will remain.
+    ///
+    /// The number of occurrences of the value currently in the bag is returned.
+    ///
+    /// The value may be any borrowed form of the bag's value type, but
+    /// [`Hash`] and [`Eq`] on the borrowed form *must* match those for
+    /// the value type.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use hashbag::HashBag;
+    ///
+    /// let mut bag = HashBag::new();
+    ///
+    /// bag.insert_many('x', 10);
+    /// assert_eq!(bag.contains(&'x'), 10);
+    /// assert_eq!(bag.remove_many(&'x', 3), 7);
+    /// assert_eq!(bag.contains(&'x'), 7);
+    /// assert_eq!(bag.remove_many(&'x', 10), 0);
+    /// ```
+    #[inline]
+    pub fn remove_many<Q: ?Sized>(&mut self, value: &Q, quantity: usize) -> usize
```

Comment:
In the interest of future-proofing and aligning with `std`, what do you think of calling this `saturating_remove_many`?

Reply:
We could, but most "saturating" methods (that I'm aware of) are on primitives, so it feels out of place here. Moreover, a "saturating" method implies the existence of an "will panic on under/overflow" method, which also doesn't exist and would be out of place if it did.

Reply:
Yeah, it is a slight abuse of the term. How about `remove_up_to`?

Alternatively, what do you think of using `removen` (like `std`'s `splitn`) instead of `remove_many`? I believe `splitn` already functions such that if you have fewer elements than `n`, it won't error, it just won't give you more splits than there are (bounded by `n`).

I don't have a problem with the implication of the existence of a method that doesn't actually exist — after all, that's the decision we're making here; to only add the method that's actually needed right now.

Reply:
I am a fan of `remove_up_to`. It's clear (unlike `remove_many`) and concise. Also, it doesn't read as though there might be a complementary method (like `saturating_remove_many` does).